core/src/main/java/org/infinispan/util/concurrent/BlockingManagerImpl.java

```diff
@@ -197,6 +197,7 @@ public void execute(Runnable command) {
             log.tracef("Invoked thenRun on a blocking thread, joining %s in same blocking thread", traceId);
          }
          try {
+            CompletionStages.join(stage);
             return stage.thenRun(runnable);
```

Comment:
it isn't trivial for me xD
Why do you need `stage.thenRun(runnable)`? 
Could you just invoke `runnable.run()` and `return CompletableFutures.completedNull()`?

Reply:
Sure, either way would work. The point is the stage needs to be complete. I can change it to that if you want.

Reply:
I would say to invoke the `Runnable` directly.
`java.util.concurrent.CompletableFuture#thenRun()` seems to create a new `CompletableFuture` instance.

Reply:
Yep, I already pushed the change. 👍